src/main/java/com/cloudbees/hudson/plugins/folder/properties/FolderCredentialsProvider.java

```diff
@@ -69,17 +61,38 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.WeakHashMap;
+import jenkins.model.Jenkins;
 import jenkins.model.TransientActionFactory;
+import net.jcip.annotations.GuardedBy;
+import net.sf.json.JSONObject;
+import org.acegisecurity.Authentication;
+import org.acegisecurity.context.SecurityContext;
+import org.acegisecurity.context.SecurityContextHolder;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
 
 /**
  * A store of credentials that can be used as a Stapler opbject.
  */
 @Extension(optional = true)
 public class FolderCredentialsProvider extends CredentialsProvider {
 
+    /**
+     * The valid scopes for this store.
+     */
     private static final Set<CredentialsScope> SCOPES =
             Collections.<CredentialsScope>singleton(CredentialsScope.GLOBAL);
 
+    @GuardedBy("self")
+    private static final WeakHashMap<AbstractFolder<?>,FolderCredentialsProperty> emptyProperties =
+            new WeakHashMap<AbstractFolder<?>, FolderCredentialsProperty>();
```

Comment:
IIUC this is designed to solve a problem analogous to the proprietary CJP-1758, correct? To wit: merely saving a folder configuration should not add an empty `FolderCredentialsProperty` to the `config.xml`; this should only be created if and when some credentials are actually added to it.